src/code/z_bg_item.c

```diff
@@ -1,99 +1,144 @@
 #include "global.h"
 
-void BcCheck3_BgActorInit(DynaPolyActor* actor, UNK_TYPE4 param_2) {
-    actor->bgId = -1;
-    actor->unk148 = 0;
-    actor->unk14C = 0;
-    actor->unk154 = param_2;
-    actor->unk_158 = 0;
+#define DYNAPOLY_STATE_NONE 0
+#define DYNAPOLY_STATE_RIDING_FALLING 1 << 0
+#define DYNAPOLY_STATE_RIDING_MOVING 1 << 1
+#define DYNAPOLY_STATE_RIDING_ROTATING 1 << 2
+#define DYNAPOLY_STATE_SWITCH_PRESSED 1 << 3
+#define DYNAPOLY_STATE_HEAVY_SWITCH_PRESSED 1 << 4
+
+void DynaPoly_Init(DynaPolyActor* dynaActor, s32 flags) {
+    dynaActor->bgId = -1;
+    dynaActor->pushForce = 0.0f;
+    dynaActor->unk14C = 0.0f;
+    dynaActor->flags = flags;
+    dynaActor->stateFlags = DYNAPOLY_STATE_NONE;
 }
 
-void BgCheck3_LoadMesh(GlobalContext* globalCtx, DynaPolyActor* actor, CollisionHeader* meshHeader) {
-    CollisionHeader* header;
+void DynaPoly_LoadMesh(GlobalContext* globalCtx, DynaPolyActor* dynaActor, CollisionHeader* meshHeader) {
+    CollisionHeader* header = NULL;
 
     header = NULL;
     BgCheck_RelocateMeshHeader(meshHeader, &header);
-    actor->bgId = BgCheck_AddActorMesh(globalCtx, &globalCtx->colCtx.dyna, actor, header);
+    dynaActor->bgId = BgCheck_AddActorMesh(globalCtx, &globalCtx->colCtx.dyna, dynaActor, header);
 }
 
-void BgCheck3_ResetFlags(DynaPolyActor* actor) {
-    actor->unk_158 = 0;
+void DynaPoly_ResetState(DynaPolyActor* dynaActor) {
+    dynaActor->stateFlags = DYNAPOLY_STATE_NONE;
 }
 
-void func_800CAE88(DynaPolyActor* actor) {
-    actor->unk_158 |= 1;
+void DynaPoly_SetRidingFallingState(DynaPolyActor* dynaActor) {
+    dynaActor->stateFlags |= DYNAPOLY_STATE_RIDING_FALLING;
 }
 
-void func_800CAE9C(DynaPolyActor* actor) {
-    actor->unk_158 |= 2;
+void DynaPoly_SetRidingMovingState(DynaPolyActor* dynaActor) {
+    dynaActor->stateFlags |= DYNAPOLY_STATE_RIDING_MOVING;
 }
 
-void func_800CAEB0(CollisionContext* colCtx, s32 index) {
-    DynaPolyActor* actor;
+void DynaPoly_SetRidingMovingStateByIndex(CollisionContext* colCtx, s32 index) {
+    DynaPolyActor* dynaActor;
```

Comment:
```suggestion
    DynaPolyActor* dynaActor = BgCheck_GetActorOfMesh(colCtx, index);
```
Could set dynaActor right away.